arangod/Aql/QueryRegistry.h

```diff
@@ -71,16 +75,18 @@ class QueryRegistry {
   /// Note that an open query will not expire, so users should please
   /// protect against leaks. If an already open query is found, an exception
   /// is thrown.
-  void* openEngine(EngineId eid, EngineType type);
-  ExecutionEngine* openExecutionEngine(EngineId eid) {
-    return static_cast<ExecutionEngine*>(
-        openEngine(eid, EngineType::Execution));
+  ResultT<void*> openEngine(EngineId eid, EngineType type,
```

Comment:
Not in this PR.
But as far as I see openEngine really wants to be a template
openEngnie<EngineType>
so it guarantees to return the correct type and not void*.
We have the wrappers anyhow.

Reply:
Yes, but then the behavior of `openGraphEngine` must be consistent (it currently isn't). I was planning to do this when investigating how to use parallel gather with traversals (which is currently disabled).